src/main/java/com/commercetools/sync/services/CartDiscountService.java

```diff
@@ -0,0 +1,80 @@
+package com.commercetools.sync.services;
+
+import io.sphere.sdk.cartdiscounts.CartDiscount;
+import io.sphere.sdk.cartdiscounts.CartDiscountDraft;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.SphereException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletionStage;
+
+public interface CartDiscountService {
+    /**
+     * Given a {@link Set} of CartDiscount keys, this method fetches a set of all the CartDiscounts, matching this given
+     * set of keys in the CTP project, defined in an injected {@link SphereClient}. A
+     * mapping of the key to the id of the fetched CartDiscount is persisted in an in-memory map.
```

Comment:
A bit confusing why you mention `mapping of the key to the id` here as it's not used for anything in this class? At first I thought you're describing that this method is returning that map.

Reply:
but it is used. The `keyToIdCache`